Packetery/Checkout/Model/Carrier/Imp/Packetery/Brain.php

```diff
@@ -106,4 +123,111 @@ public function getAvailableCountries(array $methods): array {
     public function getBaseCountries(): array {
         return ['CZ', 'SK', 'HU', 'RO'];
     }
+
+    public function getDynamicCarrierById( ?int $id ): ?\Packetery\Checkout\Model\Carrier\AbstractDynamicCarrier {
+        if ($id === null) {
+            return null;
+        }
+
+        foreach ( $this->findResolvableDynamicCarriers() as $carrier ) {
+            if ($id === $carrier->getCarrierId()) {
+                return $carrier;
+            }
+        }
+
+        return null;
+    }
+
+    public function validateDynamicCarrier( string $method, string $countryId, ?\Packetery\Checkout\Model\Carrier\AbstractDynamicCarrier $dynamicCarrier = null ): bool {
+        if ($dynamicCarrier === null) {
+            return true;
+        }
+
+        if ($dynamicCarrier->getCountryId() === $countryId && in_array($method, $dynamicCarrier->getMethods(), true)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @return \Packetery\Checkout\Model\Carrier\AbstractDynamicCarrier[]
+     */
+    public function findResolvableDynamicCarriers(): array {
+        $zpointTitle = 'Packeta internal pickup points';
+        $zboxTitle = 'Packeta - Z-BOXes';
```

Comment:
Nemělo by i zde být jednotné číslo?

Reply:
Ty jsi měl ve WP "Z-BOXes". Takže finální znění je "Packeta - Z-BOX" ?

Reply:
J. si vyžádal jednotné číslo, vidím, že i ty jsi to na některých místech měnil.